types/block.go

```diff
@@ -801,6 +805,8 @@ func (commit *Commit) GetVote(valIdx int32) *Vote {
 		ValidatorAddress: commitSig.ValidatorAddress,
 		ValidatorIndex:   valIdx,
 		Signature:        commitSig.Signature,
+		// XXX: check if self-authenticating is needed
```

Comment:
What does this mean?

Reply:
Whether we need self authenticating data inside the Vote struct or not. Once a vote is verified and included in the vote pool, we don't need self-auth inside the Vote struct, but I don't know where the `GetVote` is called, should check it. We could not derive a Vote with self-auth data from Commit, because Commit discards the data.